xbmc/utils/Map.h

```diff
@@ -0,0 +1,102 @@
+
+/*
+ *  Copyright (C) 2005-2022 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#pragma once
+
+#include <algorithm>
+#include <array>
+#include <stdexcept>
+
+/*!
+ * \brief This class is designed to implement a constexpr version of std::map.
+ *        The standard library std::map doesn't allow constexpr (and it
+ *        doesn't look like it will be implemented in the future). This class
+ *        utilizes std::array and std::pair as they allow constexpr.
+ *
+ *        When using this class you should use the helper make_map instead of
+ *        constructing this class directly. For example:
+ *        constexpr auto myMap = make_map<int, std::string_view>({{1, "one"}});
+ *
+ *        This class is useful for mapping enum values to strings that can be
+ *        compile time checked. This also helps with heap usage.
+ *
```

Comment:
Note that lookups have linear complexity, so should not be used for "big" maps

Reply:
I've added this to the doxy. Do you think it would be good to static_assert if the size is large than some value? I guess it may be arbitrary so the comment is likely "good enough".

Reply:
Yeah good enough for me :+1: